stable/pomerium/templates/proxy-service.yaml

```diff
@@ -11,9 +11,15 @@ metadata:
 {{- if .Values.service.labels }}
 {{ toYaml .Values.service.labels | indent 4 }}
 {{- end }}
-{{- if .Values.service.annotations }}
-  annotations:
-{{ toYaml .Values.service.annotations | indent 4 }}
+annotations:
+{{- if .Values.proxy.service.annotations }}
+ {{- range $key, $value := .Values.proxy.service.annotations }}
+   {{ $key }}: {{ $value | quote }}
+ {{- end }}
+{{- else if .Values.service.annotations }}
```

Comment:
@travisgroth why is it needed to use an if/else? I would have thought that we could directly have used both `proxy.service.annotations` and `service.annotations` if one wants to combine general annotations + service specific ones.

Reply:
Thought about that.  I figured you may want to override rather than add.  If you want both you could always bring your shared annotations into the service specific one.  You could facilitate both but that seemed like it would need too much logic for a special case.  We can always add this if it becomes painful.

Reply:
if you need to override, then you can also use the merge operator from sprig (usable in helm out of the box) to create a dict from `service.annotations` and `proxy.service.annotations`.

This would have the advantage of not having to maintain copied annotation for the user.

Reply:
Ah - I'm not worried about the merge (thought it's good to know about that sprig function).  The low tech way of enumerating both inline is easy.  

I was more thinking about strict override (current implementation) vs merge override.  In the absence of implementing both - which is the added complexity I was talking about - strict override is more flexible.  Granted, it is a little more verbose for the end user who wants merge shared and service-specific, but it does the job.  If there's prominent use cases for merge-override that I'm not considering, I'm happy to add that in.